packages/core/src/lib/input/numeric-input.svelte

```diff
@@ -0,0 +1,87 @@
+<!--
+@component
+  
+For numeric user inputs.
+
+```svelte
+<NumericInput type="integer"  on:input={onInput} />
+```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import { createEventDispatcher } from 'svelte';
+import Input from './input.svelte';
+import { getDecimals, type NumericInputTypes } from './utils';
+
+/** The input type */
+export let type: NumericInputTypes = 'number';
+
+export let value: number | undefined = undefined;
+
+export let step = 1;
+
+const dispatch = createEventDispatcher<{
+  input: number;
+  blur: void;
+  keydown: number;
+}>();
```

Comment:
nit: docstrings, please!

Reply:
This is a draft PR, I will add tests and docs later I just put this up so you can grab the state code from the input.